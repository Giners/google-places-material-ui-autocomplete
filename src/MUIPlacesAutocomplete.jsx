import React from 'react'
import Typography from 'material-ui/Typography'

const MUIPlacesAutocomplete = () => (<Typography>Hello World - Placeholder</Typography>)

export default MUIPlacesAutocomplete
