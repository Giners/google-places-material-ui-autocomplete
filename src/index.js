import MUIPlacesAutocomplete from './MUIPlacesAutocomplete'

export default MUIPlacesAutocomplete
