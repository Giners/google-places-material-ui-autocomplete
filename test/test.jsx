// Disable 'prefer-arrow-callback' as Mocha recommends against passing arrow functions to Mocha. See
// https://mochajs.org/#arrow-functions for more info.
// Disable 'func-names' so we can use anonymous functions as a convenience to us as test writers.
/* eslint prefer-arrow-callback: 0, func-names: 0 */
/* eslint no-unused-expressions: 0, chai-friendly/no-unused-expressions: 2 */

// 3rd-party supporting test libs
import chai, { expect } from 'chai'
import chaiJestSnapshot from 'chai-jest-snapshot'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
import React from 'react'

// Code under test
import { MUIPlacesAutocomplete } from './../src/MUIPlacesAutocomplete'

// Supporting code
import getACServiceClassDef from './testHelper'

// Configure Chai to work with Jest
chai.use(chaiJestSnapshot)

describe('React component test: <MUIPlacesAutocomplete>', function () {
  // Common search value that can be used throughout the tests
  const searchInputValue = 'Bellingham'

  before('Configure Enzyme', function () {
    Enzyme.configure({ adapter: new Adapter() })
  })

  before('Reset Jest snapshot registry for Chai usage', function () {
    chaiJestSnapshot.resetSnapshotRegistry()
  })

  beforeEach('Configure chai-jest-snapshot for "Mocha configuration mode"', function () {
    chaiJestSnapshot.configureUsingMochaContext(this)
  })

  describe('Renders correctly for given application state:', function () {
    // Wrapper providing access to DOM APIs/full DOM rendering for the <MUIPlacesAutocomplete>
    // component that will be under test
    let mpaWrapper = null

    before('"Load" the Google Maps JavaScript API on \'window\'', function () {
      // These tests depend on a DOM to be setup for more indepth tests that either do full DOM
      // rendering or for leveraging DOM APIs. At this point we presume that the DOM has been setup
      // and we can add additional properties to the global 'window' object.
      expect(global.window).to.exist

      // The <MUIPlacesAutocomplete> component expects the Google Maps JavaScript API to be loaded
      // in the 'window' object. Since we aren't loading it we mock our the API and add it to the
      // 'window' object.
      global.window.google = {
        maps: {
          places: {
            // We disable the ESLint rule 'class-methods-use-this' on the next line as our component
            // depends on the method being available to an instance of the AutocompleteService
            // class (i.e. we can't convert it into a static).
            // eslint-disable-next-line class-methods-use-this
            AutocompleteService: class AutocompleteService { getPlacePredictions() { } },
            // Don't forget to mock out the status' that are used by the <MUIPlacesAutocomplete>
            // component
            PlacesServiceStatus: {
              OK: 'OK',
            },
          },
        },
      }
    })

    beforeEach('Setup Enzyme wrapper', function () {
      // Since we aren't testing the MUIPlacesAutocomplete HOC (see MUI 'withStyles()') we provide
      // an empty object for the MUI styling/classes that would normally get applied so we don't try
      // to reference an undefined prop.
      mpaWrapper = mount(<MUIPlacesAutocomplete classes={{ }} />)

      expect(mpaWrapper).to.not.be.null
    })

    it('Initial state', function () {
      expect(mpaWrapper.state().value).to.exist
      expect(mpaWrapper.state().value).to.be.empty
      expect(mpaWrapper.state().suggestions).to.exist
      expect(mpaWrapper.state().suggestions).to.be.empty

      // Protect us from forgetting to update this test if we add an additional key to the
      // components state. We currently only expect the 'value' and 'suggestions' key...
      expect(Object.keys(mpaWrapper.state()).length).to.equal(2)

      // We snapshot the initial state to provide visibility to any changes we have made to our
      // component...
      expect(toJson(mpaWrapper)).to.matchSnapshot()
    })

    it('Suggestions rendered from \'suggestions\' state', function () {
      // Provide a search input and a suggestion to our state so a suggestion is rendered
      mpaWrapper.setState({
        value: searchInputValue,
        suggestions: [{ description: 'Bellingham, WA, United States' }],
      })
      mpaWrapper.find('input').simulate('focus')

      expect(mpaWrapper.state().suggestions.length).to.equal(1)
      expect(mpaWrapper.find('MenuItem').length).to.equal(mpaWrapper.state().suggestions.length)

      expect(toJson(mpaWrapper)).to.matchSnapshot()
    })

    it('Empty input renders no suggestions after previous ones rendered', function () {
      // Make sure we aren't asserting an empty list of suggestions in the first place after
      // focusing the 'input' element...
      mpaWrapper.setState({
        value: searchInputValue,
        suggestions: [{ description: 'Bellingham, WA, United States' }],
      })
      mpaWrapper.find('input').simulate('focus')

      expect(mpaWrapper.state().suggestions.length).to.equal(1)
      expect(mpaWrapper.find('MenuItem').length).to.equal(mpaWrapper.state().suggestions.length)

      // Now check that we no longer render suggestions on empty input...
      mpaWrapper.find('input').simulate('change', { target: { value: '' } })

      expect(mpaWrapper.state().value).to.be.empty
      expect(mpaWrapper.state().suggestions).to.be.empty
      expect(mpaWrapper.find('MenuItem').length).to.equal(0)

      expect(toJson(mpaWrapper)).to.matchSnapshot()
    })

    it('Google logo is present in a populated list of suggestions', function () {
      // Provide a search input and a suggestion to our state so we can render the suggestions
      // container which will have the Google Logo in a single <img> element
      mpaWrapper.setState({
        value: searchInputValue,
        suggestions: [{ description: 'Bellingham, WA, United States' }],
      })
      mpaWrapper.find('input').simulate('focus')

      expect(mpaWrapper.find('img').length).to.equal(1)

      expect(toJson(mpaWrapper)).to.matchSnapshot()
    })
  })

  describe('Consumes Google Maps JavaScript API correctly:', function () {
    it('AutocompleteService.getPlacePredictions() returns predictions for given input', function (done) {
      let mpaWrapper = null

      // Find the 'input' element so we can simulate an event for changing the input which will
      // cause our component to get place predictions with the Google Maps API and ultimately update
      // its state with place suggestions. Before doing so setup our test callback so we can assert
      // that suggestions have been populated and signal that we have completed the test.
      const testCallback = () => {
        try {
          expect(mpaWrapper.state().value).to.equal(searchInputValue)
          expect(mpaWrapper.state().suggestions).to.not.be.empty
        } catch (e) {
          done(e)
          return
        }

        done()
      }

      // This test depends on a DOM to be setup. At this point we presume that the DOM has been
      // setup and we can add additional properties to the global 'window' object.
      expect(global.window).to.exist

      // The <MUIPlacesAutocomplete> component expects the Google Maps JavaScript API to be loaded
      // in the 'window' object. Use the mock API test helper that actually wraps the Google Maps
      // JavaScript API and add it to the 'window' object. We setup our DOM here rather than in a
      // 'before' Mocha hook as we want to use the mock API with our specific test callback.
      global.window.google = {
        maps: {
          places: {
            AutocompleteService: getACServiceClassDef(testCallback),
            // Don't forget to mock out the status' that are used by the <MUIPlacesAutocomplete>
            // component
            PlacesServiceStatus: {
              OK: 'OK',
            },
          },
        },
      }

      // Now mount our component to make use of our mock API when we send a 'change' event on our
      // 'input' element
      mpaWrapper = mount(<MUIPlacesAutocomplete classes={{ }} />)

      const inputWrapper = mpaWrapper.find('input')

      inputWrapper.simulate('change', { target: { value: searchInputValue } })
    })
  })
})
