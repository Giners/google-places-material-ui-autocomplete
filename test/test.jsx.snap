// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`React component test: <MUIPlacesAutocomplete> Renders correctly for given application state: Duplicate suggestions aren't rendered 1`] = `
<MenuList>
  <WithStyles(List)
    className={undefined}
    onBlur={[Function]}
    onKeyDown={[Function]}
    role="menu"
  >
    <List
      className={undefined}
      classes={
        Object {
          "dense": "MuiList-dense-48",
          "padding": "MuiList-padding-47",
          "root": "MuiList-root-46",
          "subheader": "MuiList-subheader-49",
        }
      }
      component="ul"
      dense={false}
      disablePadding={false}
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="menu"
    >
      <ul
        className="MuiList-root-46 MuiList-padding-47"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="menu"
      >
        <WithStyles(MenuItem)
          component="div"
          id="downshift-3-item-0"
          key=".0:$Bellingham, WA, United States"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          selected={false}
          tabIndex={-1}
        >
          <MenuItem
            classes={
              Object {
                "root": "MuiMenuItem-root-50",
                "selected": "MuiMenuItem-selected-51",
              }
            }
            component="div"
            id="downshift-3-item-0"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            selected={false}
            tabIndex={-1}
          >
            <WithStyles(ListItem)
              button={true}
              className="MuiMenuItem-root-50"
              component="div"
              id="downshift-3-item-0"
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              <ListItem
                ContainerComponent="li"
                button={true}
                className="MuiMenuItem-root-50"
                classes={
                  Object {
                    "button": "MuiListItem-button-60",
                    "container": "MuiListItem-container-53",
                    "default": "MuiListItem-default-55",
                    "dense": "MuiListItem-dense-56",
                    "disabled": "MuiListItem-disabled-57",
                    "divider": "MuiListItem-divider-58",
                    "gutters": "MuiListItem-gutters-59",
                    "keyboardFocused": "MuiListItem-keyboardFocused-54",
                    "root": "MuiListItem-root-52",
                    "secondaryAction": "MuiListItem-secondaryAction-61",
                  }
                }
                component="div"
                dense={false}
                disableGutters={false}
                disabled={false}
                divider={false}
                id="downshift-3-item-0"
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                <WithStyles(ButtonBase)
                  className="MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                  classes={
                    Object {
                      "keyboardFocused": "MuiListItem-keyboardFocused-54",
                    }
                  }
                  component="div"
                  disabled={false}
                  id="downshift-3-item-0"
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseMove={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  <ButtonBase
                    centerRipple={false}
                    className="MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                    classes={
                      Object {
                        "disabled": "MuiButtonBase-disabled-63",
                        "keyboardFocused": "MuiButtonBase-keyboardFocused-64 MuiListItem-keyboardFocused-54",
                        "root": "MuiButtonBase-root-62",
                      }
                    }
                    component="div"
                    disableRipple={false}
                    disabled={false}
                    focusRipple={false}
                    id="downshift-3-item-0"
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="menuitem"
                    tabIndex={-1}
                    type="button"
                  >
                    <div
                      className="MuiButtonBase-root-62 MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                      id="downshift-3-item-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseMove={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="menuitem"
                      tabIndex={-1}
                    >
                      <div
                        key=".0"
                      >
                        <strong
                          key="0"
                          style={
                            Object {
                              "fontWeight": 500,
                            }
                          }
                        >
                          Bellingham
                        </strong>
                        <span
                          key="1"
                          style={
                            Object {
                              "fontWeight": 300,
                            }
                          }
                        >
                          , WA, United States
                        </span>
                      </div>
                      <WithStyles(TouchRipple)
                        center={false}
                        innerRef={[Function]}
                      >
                        <TouchRipple
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child-69",
                              "childLeaving": "MuiTouchRipple-childLeaving-70",
                              "childPulsate": "MuiTouchRipple-childPulsate-71",
                              "ripple": "MuiTouchRipple-ripple-66",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                              "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                              "root": "MuiTouchRipple-root-65",
                            }
                          }
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            className="MuiTouchRipple-root-65"
                            component="span"
                            enter={true}
                            exit={true}
                          >
                            <span
                              className="MuiTouchRipple-root-65"
                            />
                          </TransitionGroup>
                        </TouchRipple>
                      </WithStyles(TouchRipple)>
                    </div>
                  </ButtonBase>
                </WithStyles(ButtonBase)>
              </ListItem>
            </WithStyles(ListItem)>
          </MenuItem>
        </WithStyles(MenuItem)>
        <div
          key=".1"
          onFocus={[Function]}
          style={
            Object {
              "display": "flex",
            }
          }
          tabIndex={-1}
        >
          <span
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <img
            alt=""
            src={Object {}}
          />
        </div>
      </ul>
    </List>
  </WithStyles(List)>
</MenuList>
`;

exports[`React component test: <MUIPlacesAutocomplete> Renders correctly for given application state: Google logo is present in a populated list of suggestions 1`] = `
<MenuList>
  <WithStyles(List)
    className={undefined}
    onBlur={[Function]}
    onKeyDown={[Function]}
    role="menu"
  >
    <List
      className={undefined}
      classes={
        Object {
          "dense": "MuiList-dense-48",
          "padding": "MuiList-padding-47",
          "root": "MuiList-root-46",
          "subheader": "MuiList-subheader-49",
        }
      }
      component="ul"
      dense={false}
      disablePadding={false}
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="menu"
    >
      <ul
        className="MuiList-root-46 MuiList-padding-47"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="menu"
      >
        <WithStyles(MenuItem)
          component="div"
          id="downshift-4-item-0"
          key=".0:$Bellingham, WA, United States"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          selected={false}
          tabIndex={-1}
        >
          <MenuItem
            classes={
              Object {
                "root": "MuiMenuItem-root-50",
                "selected": "MuiMenuItem-selected-51",
              }
            }
            component="div"
            id="downshift-4-item-0"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            selected={false}
            tabIndex={-1}
          >
            <WithStyles(ListItem)
              button={true}
              className="MuiMenuItem-root-50"
              component="div"
              id="downshift-4-item-0"
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              <ListItem
                ContainerComponent="li"
                button={true}
                className="MuiMenuItem-root-50"
                classes={
                  Object {
                    "button": "MuiListItem-button-60",
                    "container": "MuiListItem-container-53",
                    "default": "MuiListItem-default-55",
                    "dense": "MuiListItem-dense-56",
                    "disabled": "MuiListItem-disabled-57",
                    "divider": "MuiListItem-divider-58",
                    "gutters": "MuiListItem-gutters-59",
                    "keyboardFocused": "MuiListItem-keyboardFocused-54",
                    "root": "MuiListItem-root-52",
                    "secondaryAction": "MuiListItem-secondaryAction-61",
                  }
                }
                component="div"
                dense={false}
                disableGutters={false}
                disabled={false}
                divider={false}
                id="downshift-4-item-0"
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                <WithStyles(ButtonBase)
                  className="MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                  classes={
                    Object {
                      "keyboardFocused": "MuiListItem-keyboardFocused-54",
                    }
                  }
                  component="div"
                  disabled={false}
                  id="downshift-4-item-0"
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseMove={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  <ButtonBase
                    centerRipple={false}
                    className="MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                    classes={
                      Object {
                        "disabled": "MuiButtonBase-disabled-63",
                        "keyboardFocused": "MuiButtonBase-keyboardFocused-64 MuiListItem-keyboardFocused-54",
                        "root": "MuiButtonBase-root-62",
                      }
                    }
                    component="div"
                    disableRipple={false}
                    disabled={false}
                    focusRipple={false}
                    id="downshift-4-item-0"
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="menuitem"
                    tabIndex={-1}
                    type="button"
                  >
                    <div
                      className="MuiButtonBase-root-62 MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                      id="downshift-4-item-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseMove={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="menuitem"
                      tabIndex={-1}
                    >
                      <div
                        key=".0"
                      >
                        <strong
                          key="0"
                          style={
                            Object {
                              "fontWeight": 500,
                            }
                          }
                        >
                          Bellingham
                        </strong>
                        <span
                          key="1"
                          style={
                            Object {
                              "fontWeight": 300,
                            }
                          }
                        >
                          , WA, United States
                        </span>
                      </div>
                      <WithStyles(TouchRipple)
                        center={false}
                        innerRef={[Function]}
                      >
                        <TouchRipple
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child-69",
                              "childLeaving": "MuiTouchRipple-childLeaving-70",
                              "childPulsate": "MuiTouchRipple-childPulsate-71",
                              "ripple": "MuiTouchRipple-ripple-66",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                              "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                              "root": "MuiTouchRipple-root-65",
                            }
                          }
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            className="MuiTouchRipple-root-65"
                            component="span"
                            enter={true}
                            exit={true}
                          >
                            <span
                              className="MuiTouchRipple-root-65"
                            />
                          </TransitionGroup>
                        </TouchRipple>
                      </WithStyles(TouchRipple)>
                    </div>
                  </ButtonBase>
                </WithStyles(ButtonBase)>
              </ListItem>
            </WithStyles(ListItem)>
          </MenuItem>
        </WithStyles(MenuItem)>
        <div
          key=".1"
          onFocus={[Function]}
          style={
            Object {
              "display": "flex",
            }
          }
          tabIndex={-1}
        >
          <span
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <img
            alt=""
            src={Object {}}
          />
        </div>
      </ul>
    </List>
  </WithStyles(List)>
</MenuList>
`;

exports[`React component test: <MUIPlacesAutocomplete> Renders correctly for given application state: Suggestions rendered from 'suggestions' state 1`] = `
<MenuList>
  <WithStyles(List)
    className={undefined}
    onBlur={[Function]}
    onKeyDown={[Function]}
    role="menu"
  >
    <List
      className={undefined}
      classes={
        Object {
          "dense": "MuiList-dense-48",
          "padding": "MuiList-padding-47",
          "root": "MuiList-root-46",
          "subheader": "MuiList-subheader-49",
        }
      }
      component="ul"
      dense={false}
      disablePadding={false}
      onBlur={[Function]}
      onKeyDown={[Function]}
      role="menu"
    >
      <ul
        className="MuiList-root-46 MuiList-padding-47"
        onBlur={[Function]}
        onKeyDown={[Function]}
        role="menu"
      >
        <WithStyles(MenuItem)
          component="div"
          id="downshift-1-item-0"
          key=".0:$Bellingham, WA, United States"
          onClick={[Function]}
          onFocus={[Function]}
          onMouseDown={[Function]}
          onMouseMove={[Function]}
          selected={false}
          tabIndex={-1}
        >
          <MenuItem
            classes={
              Object {
                "root": "MuiMenuItem-root-50",
                "selected": "MuiMenuItem-selected-51",
              }
            }
            component="div"
            id="downshift-1-item-0"
            onClick={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            onMouseMove={[Function]}
            role="menuitem"
            selected={false}
            tabIndex={-1}
          >
            <WithStyles(ListItem)
              button={true}
              className="MuiMenuItem-root-50"
              component="div"
              id="downshift-1-item-0"
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              onMouseMove={[Function]}
              role="menuitem"
              tabIndex={-1}
            >
              <ListItem
                ContainerComponent="li"
                button={true}
                className="MuiMenuItem-root-50"
                classes={
                  Object {
                    "button": "MuiListItem-button-60",
                    "container": "MuiListItem-container-53",
                    "default": "MuiListItem-default-55",
                    "dense": "MuiListItem-dense-56",
                    "disabled": "MuiListItem-disabled-57",
                    "divider": "MuiListItem-divider-58",
                    "gutters": "MuiListItem-gutters-59",
                    "keyboardFocused": "MuiListItem-keyboardFocused-54",
                    "root": "MuiListItem-root-52",
                    "secondaryAction": "MuiListItem-secondaryAction-61",
                  }
                }
                component="div"
                dense={false}
                disableGutters={false}
                disabled={false}
                divider={false}
                id="downshift-1-item-0"
                onClick={[Function]}
                onFocus={[Function]}
                onMouseDown={[Function]}
                onMouseMove={[Function]}
                role="menuitem"
                tabIndex={-1}
              >
                <WithStyles(ButtonBase)
                  className="MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                  classes={
                    Object {
                      "keyboardFocused": "MuiListItem-keyboardFocused-54",
                    }
                  }
                  component="div"
                  disabled={false}
                  id="downshift-1-item-0"
                  onClick={[Function]}
                  onFocus={[Function]}
                  onMouseDown={[Function]}
                  onMouseMove={[Function]}
                  role="menuitem"
                  tabIndex={-1}
                >
                  <ButtonBase
                    centerRipple={false}
                    className="MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                    classes={
                      Object {
                        "disabled": "MuiButtonBase-disabled-63",
                        "keyboardFocused": "MuiButtonBase-keyboardFocused-64 MuiListItem-keyboardFocused-54",
                        "root": "MuiButtonBase-root-62",
                      }
                    }
                    component="div"
                    disableRipple={false}
                    disabled={false}
                    focusRipple={false}
                    id="downshift-1-item-0"
                    onClick={[Function]}
                    onFocus={[Function]}
                    onMouseDown={[Function]}
                    onMouseMove={[Function]}
                    role="menuitem"
                    tabIndex={-1}
                    type="button"
                  >
                    <div
                      className="MuiButtonBase-root-62 MuiListItem-root-52 MuiListItem-default-55 MuiListItem-gutters-59 MuiListItem-button-60 MuiMenuItem-root-50"
                      id="downshift-1-item-0"
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseMove={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      role="menuitem"
                      tabIndex={-1}
                    >
                      <div
                        key=".0"
                      >
                        <strong
                          key="0"
                          style={
                            Object {
                              "fontWeight": 500,
                            }
                          }
                        >
                          Bellingham
                        </strong>
                        <span
                          key="1"
                          style={
                            Object {
                              "fontWeight": 300,
                            }
                          }
                        >
                          , WA, United States
                        </span>
                      </div>
                      <WithStyles(TouchRipple)
                        center={false}
                        innerRef={[Function]}
                      >
                        <TouchRipple
                          center={false}
                          classes={
                            Object {
                              "child": "MuiTouchRipple-child-69",
                              "childLeaving": "MuiTouchRipple-childLeaving-70",
                              "childPulsate": "MuiTouchRipple-childPulsate-71",
                              "ripple": "MuiTouchRipple-ripple-66",
                              "ripplePulsate": "MuiTouchRipple-ripplePulsate-68",
                              "rippleVisible": "MuiTouchRipple-rippleVisible-67",
                              "root": "MuiTouchRipple-root-65",
                            }
                          }
                        >
                          <TransitionGroup
                            childFactory={[Function]}
                            className="MuiTouchRipple-root-65"
                            component="span"
                            enter={true}
                            exit={true}
                          >
                            <span
                              className="MuiTouchRipple-root-65"
                            />
                          </TransitionGroup>
                        </TouchRipple>
                      </WithStyles(TouchRipple)>
                    </div>
                  </ButtonBase>
                </WithStyles(ButtonBase)>
              </ListItem>
            </WithStyles(ListItem)>
          </MenuItem>
        </WithStyles(MenuItem)>
        <div
          key=".1"
          onFocus={[Function]}
          style={
            Object {
              "display": "flex",
            }
          }
          tabIndex={-1}
        >
          <span
            style={
              Object {
                "flex": 1,
              }
            }
          />
          <img
            alt=""
            src={Object {}}
          />
        </div>
      </ul>
    </List>
  </WithStyles(List)>
</MenuList>
`;
